### ISSUE TITLE
Default constructor is required when not instantiating

### ISSUE THREAD

**Issue by bzuillsmith** [expressmapper/ExpressMapper#76]
I tried ```mapper.Map(this, myObject)``` and ExpressMapper complains that the class for the ```this``` object does not have a default constructor. I could be wrong, but couldn't we skip that requirement when we are not instantiating?

**Comment by anisimovyuriy** [comment]
Hi Ben,

From my experience :) -  there  are no checks for source object at all, but there are checks for "destination" for sure even you supply instance of destination, still there could be other nested object type properties that should be created. 
Don't you mind to send me more detailed error?

Thanks a lot!
Yuriy

**Comment by bzuillsmith** [comment]
Sorry, it was actually a call that looked like ```_mapper.Map(myObject, this);```. The destination object does not have a default constructor. Why does it need one?

```
   at System.Linq.Expressions.Expression.New(Type type)
   at ExpressMapper.TypeMapperBase`2.GetDestionationVariable() in C:\SourceCode\ExpressMapper\ExpressMapper NET40\TypeMapperBase.cs:line 176
   at ExpressMapper.DestinationTypeMapper`2.CompileInternal() in C:\SourceCode\ExpressMapper\ExpressMapper NET40\DestinationTypeMapper.cs:line 53
   at ExpressMapper.TypeMapperBase`2.Compile() in C:\SourceCode\ExpressMapper\ExpressMapper NET40\TypeMapperBase.cs:line 67
   at ExpressMapper.TypeMapperBase`2.MapTo(T src, TN dest) in C:\SourceCode\ExpressMapper\ExpressMapper NET40\TypeMapperBase.cs:line 243
   at ExpressMapper.MappingServiceProvider.MapInternal[T,TN](T src, TN dest, Boolean dynamicTrial) in C:\SourceCode\ExpressMapper\ExpressMapper NET40\MappingServiceProvider.cs:line 273
   at ExpressMapper.MappingServiceProvider.MapInternal[T,TN](T src, TN dest, Boolean dynamicTrial) in C:\SourceCode\ExpressMapper\ExpressMapper NET40\MappingServiceProvider.cs:line 299
   at ExpressMapper.MappingServiceProvider.Map[T,TN](T src, TN dest) in C:\SourceCode\ExpressMapper\ExpressMapper NET40\MappingServiceProvider.cs:line 246
   at GdsEstimating.WinBidPro.ViewModels.DoorFrameViewModel.SetFromDoorFrame(DoorFrame doorFrame) in 
   ...
```

**Comment by anisimovyuriy** [comment]
Hi Ben thx a lot for your input - actually the day before yesterday I fixed this generic <a href="https://github.com/expressmapper/ExpressMapper/issues/75">error</a> to understandable format. It happens when you try to map source to destination and destination has object type properties without default constructor (With parameters) and it happens when you don't register mapping for it (Nested property) - so it is dynamically mapped. Anyway I'm going to release new version today or if you in a big hurry you can download recent master branch and try it.

Thanks for your support,
Yuriy

**Comment by anisimovyuriy** [comment]
Fixed and will be included in Release 1.8
